src/highdicom/legacy/sop.py

```diff
@@ -25,384 +36,2630 @@
+            'InstitutionalDepartmentName',
+            'ManufacturerModelName',
+            'DeviceSerialNumber',
+            'SoftwareVersions',
+            'GantryID',
+            'PixelPaddingValue',
+            'Modality',
+            'ImageType',
+            'BurnedInAnnotation',
+            'SOPClassUID',
+            'Rows',
+            'Columns',
+            'BitsStored',
+            'BitsAllocated',
+            'HighBit',
+            'PixelRepresentation',
+            'PhotometricInterpretation',
+            'PlanarConfiguration',
+            'SamplesPerPixel',
+            'ProtocolName',
+            'ImageOrientationPatient',
+            'PixelSpacing',
+            'SliceThickness',
+            'AcquisitionContextSequence']
+        to_be_removed_from_distinguishing_attribs: set = set()
+        self._frame_sets: List[FrameSet] = []
+        frame_counts = []
+        frameset_counter = 0
+        while len(self.mixed_frames_copy) != 0:
+            frameset_counter += 1
+            x = self._find_all_similar_to_first_datasets()
+            self._frame_sets.append(FrameSet(x[0], x[1]))
+            frame_counts.append(len(x[0]))
+            # log information
+            logger.debug("Frameset({:02d}) including {:03d} frames".format(
+                frameset_counter, len(x[0])))
+            logger.debug('\t Distinguishing tags:')
+            for dg_i, dg_tg in enumerate(x[1], 1):
+                logger.debug('\t\t{:02d}/{})\t{}-{:32.32s} = {:32.32s}'.format(
+                    dg_i, len(x[1]), ـDicomHelper.tag2str(dg_tg),
+                    keyword_for_tag(dg_tg),
+                    str(x[0][0][dg_tg].value)))
+            logger.debug('\t dicom datasets in this frame set:')
+            for dicom_i, dicom_ds in enumerate(x[0], 1):
+                logger.debug('\t\t{}/{})\t {}'.format(
+                    dicom_i, len(x[0]), dicom_ds['SOPInstanceUID']))
+        frames = ''
+        for i, f_count in enumerate(frame_counts, 1):
+            frames += '{: 2d}){:03d}\t'.format(i, f_count)
+        frames = '{: 2d} frameset(s) out of all {: 3d} instances:'.format(
+            len(frame_counts), len(self.mixed_frames)) + frames
+        logger.info(frames)
+        for kw in to_be_removed_from_distinguishing_attribs:
+            self.distinguishing_attribute_keywords.remove(kw)
+        self._excluded_from_perframe_tags = {}
+        for kwkw in self.distinguishing_attribute_keywords:
+            self._excluded_from_perframe_tags[tag_for_keyword(kwkw)] = False
+        self._excluded_from_perframe_tags[
+            tag_for_keyword('AcquisitionDateTime')] = False
+        self._excluded_from_perframe_tags[
+            tag_for_keyword('AcquisitionDate')] = False
+        self._excluded_from_perframe_tags[
+            tag_for_keyword('AcquisitionTime')] = False
+        self.excluded_from_functional_groups_tags = {
+            tag_for_keyword('SpecificCharacterSet'): False}
```

Comment:
Seems like this could be written much more succinctly:
```suggestion
        self._excluded_from_perframe_tags = {
            tag_for_keyword(kwkw): False for kwkw in self.distinguishing_attribute_keywords
        }
        special_exclusions = [
            'AcquisitionDateTime',
            'AcquisitionDate',
            'AcquisitionTime',
            'SpecificCharacterSet'
        ]
        for kwkw in special_exclusions:
            self._excluded_from_perframe_tags[tag_for_keyword(kwkw)] = False
```